### PR TITLE
feat(geo-widget): backend storage implementation for geo widget

### PR DESCRIPTION
**Pull request by dubsky**
This a first part of a two phase implementation of UI geo/map widget.  This PR contains updates to swagger.yml (additions only,  non breaking changes) to allow saving of the widget configuration. 'dashboard.go' serialization configuration has been updated accordingly.

- [ ] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [ ] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [ ] Rebased/mergeable
- [ ] Tests pass
- [X] http/swagger.yml updated (if modified Go structs or API)
- [ ] Feature flagged (if modified API)
- [ ] Documentation updated or issue created (provide link to issue/pr)
- [ ] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

File: http/swagger.yml
Comment (hoorayimhelping) [inline http/swagger.yml:9656]:
```suggestion
          enum: [heatmap, circleMap, pointMap, trackMap]
```

**Comment by russorat** [comment]
@stuartcarnie could you take a look at the swagger changes?

**Comment by aanthony1243** [comment]
I added @GeorgeMac as a reviewer, please wait for his response before merging.

File: http/swagger.yml
Comment (glinton) [inline http/swagger.yml:9652]: where are `field` and `colors` defined?

File: dashboard.go
Comment (GeorgeMac) [inline dashboard.go:448]: Im not super familiar with these models, but it appears there are some view marshalling tests:
`dashboard_test.go`
Could we add cases here to account for the new types? Just for good measure.

Comment (dubsky) [reply dashboard.go:448]: Fixed, haven't noticed this since swagger generators do not complain about this problem.

File: http/swagger.yml
Comment (dubsky) [reply http/swagger.yml:9652]: Fixed, haven't noticed this since swagger generators do not complain about this problem.

File: dashboard.go
Comment (glinton) [inline dashboard.go:451]: feel free to add a similar prefix to the returned error, but remove this line

**Review by glinton — CHANGES_REQUESTED** [review]
Don't forget to update `pkger` as well so this view type can be used in templates. iirc, common pkger files updated for a view type are `clone_resource`, `parser`, `parser_models`, and the parser/service_tests with an added testdata test. @TCL735 is a pro at these changes, you can look at his merged prs for an example.

**Review by glinton — APPROVED** [review]
Thanks for making those additions, sorry for the delay on the re-review

**Comment by ivankudibal** [comment]
rebase of influxdata:master was done manually... A rebase with master of the fork, as suggested by github, is not preferred

File: dashboard_test.go
Comment (Palakp41) [inline dashboard_test.go:25]: :question: Is there a reason we are updating tests for `xy` in this pr? This seems a bit out of scope for geo.

Comment (dubsky) [reply dashboard_test.go:25]: Yes, I did it because I have been adding extra test case specific to geo as requested. I wanted to make the source file to be more readable and make it obvious what are the individual cases testing. This individual addition means that you will get 'xy' or 'geo' serialization failed message if something goes wrong.

**Comment by danxmoran** [comment]
CI will probably fail until #20506 merges and this PR is updated to include it. It's under review now.

**Pull request merged**
